### DISCUSSION TITLE
Moonraker can't connect to Klipper

### DISCUSSION THREAD

**Discussion by HemlockMimosa** [bassamanator/Sovol-SV06-firmware#16] [General]
Hi folks! Wondering if you can help me.

I installed Klipper on my Pi using KIAUH, then followed instructions for flashing Klipper onto the SV06, all of which seem to have worked. 

sudo service klipper status returns the following:

klipper.service - Klipper 3D Printer Firmware SV1
     Loaded: loaded (/etc/systemd/system/klipper.service; enabled; vendor prese>
     Active: active (running) since Wed 2023-03-22 01:15:53 GMT; 9min ago
       Docs: https://www.klipper3d.org/
   Main PID: 965 (python)
      Tasks: 2 (limit: 3933)
        CPU: 1.690s
     CGroup: /system.slice/klipper.service
             └─965 /home/andre/klippy-env/bin/python /home/andre/klipper/klippy>

Mar 22 01:15:53 raspberrypi systemd[1]: Started Klipper 3D Printer Firmware SV1.
lines 1-11/11 (END)
● klipper.service - Klipper 3D Printer Firmware SV1
     Loaded: loaded (/etc/systemd/system/klipper.service; enabled; vendor preset: enabled)
     Active: active (running) since Wed 2023-03-22 01:15:53 GMT; 9min ago
       Docs: https://www.klipper3d.org/
   Main PID: 965 (python)
      Tasks: 2 (limit: 3933)
        CPU: 1.690s
     CGroup: /system.slice/klipper.service
             └─965 /home/andre/klippy-env/bin/python /home/andre/klipper/klippy/klippy.py >

I _think_ this means that Klipper is successfully running on my printer, which means that the problem is occurring when I try to modify my configuration file.

ls -l /dev/serial/by-id/ returns the following: lrwxrwxrwx 1 root root 13 Mar 22 00:55 usb-1a86_USB_Serial-if00-port0 -> ../../ttyUSB0

I've attempted to modify my configuration file in Mainsail by adding either the entire string, "usb-1a86_USB_Serial-if00-port0 -> ../../ttyUSB0" or just "usb-1a86_USB_Serial-if00-port0"

None of these have worked. In addition, I've tried to perform the "make flash FLASH_DEVICE=/dev/serial/by-id/usb-1a86_USB_Serial-if00-port0" only to learn that there is "No rule to make target 'flash'"

Do any of you have any troubleshooting advice, or is there any step I'm missing? Thanks for your advice and comments.

**Comment by bassamanator** [comment]
Do no use `make flash`, it's not something that works with all boards.

I don't think the board is flashed. Please post out put from `ls -l /dev/serial/by-id/` and paste it with backticks, or use the code option above.

EDIT: hard to see on the device I'm using, but yeah, klipper is not flashed. Please see Flash Firmware section in readme.md, there is a link to a video as well. FAT filesystem, less than 16GB, no other files on the card.

**Reply by HemlockMimosa** [reply]
Hello and thanks for getting back to me! Here is the output you requested:

`total 0
lrwxrwxrwx 1 root root 13 Mar 22 01:08 usb-1a86_USB_Serial-if00-port0 -> ../../ttyUSB0
`
...this compels me to ask what might be a very basic question. I've only installed Klipper once before, about a year ago, so you may have to forgive my lack of understanding.

Basically, I flashed Klipper onto my printer by taking the `klipper.bin` file from the zip file containing your configuration, placing it on a blank SD card, and then powering up my printer with the SD card installed.

Should I have instead generated a .bin file by following the configuration steps listed here?

**Comment by bassamanator** [comment]
Okay let's take a few steps back. I have updated the `Flash Firmware` section, and have also compiled a new `klipper.bin` using the latest version of Klipper.

I want you to flash marlin back onto the board (included now in this repo). Then I want you to flash the latest `klipper.bin` (included now in this repo). Then let's see what happens.

Please read the `Flash Firmware` section again.

What device are you using to host klipper? RPi?

**Reply by HemlockMimosa** [reply]
Hi Bassmanator! Yes, I'm using an RPi 4 to host klipper. Also, I did as you asked, but I'm afraid nothing happened. Was flashing Marlin back onto the printer supposed to restore the original firmware/interface? Because unfortunately the screen just lit up and remained blank. After I flashed the latest `klipper.bin` I got the same output from `ls -l /dev/serial/by-id/` and Moonraker remained unable to connect to klipper.

Thanks so much for your help, by the way! I remain confident that we can resolve this.

**Reply by bassamanator** [reply]
Your flashing method is not working. Are you use you're using an appropriate microSD card? Please see flashing firmware section again.

BTW I just flashed back and for several times using the included bins. The method and firmware `.bin` files are correct.

**Reply by HemlockMimosa** [reply]
I'm using the SD card that came with the printer. I deleted all files off it, but should I reformat it as well?

**Reply by bassamanator** [reply]
Flash the marlin file included in this repo. Add a random letter at the end of the file. The extension should remain `.bin`.

**Reply by HemlockMimosa** [reply]
Okay! I may be getting somewhere.

First, I did as you asked and flashed marlin, using the SD card that came with the printer, adding a single letter to the end of the `.bin`

This worked! I was able to restore the Sovol firmware.

Next, I tried the same approach using the klipper file from your repo. This actually didn't seem to work the first time -- the printer booted to Marlin as normal. I tried again and this time it seemed to flash. The printer booted to a blank screen instead of the Sovol logo.

Then I ran `ls -l /dev/serial/by-id/` with the same result as before. Output was `total 0 lrwxrwxrwx 1 root root 13 Mar 22 01:08 usb-1a86_USB_Serial-if00-port0 -> ../../ttyUSB0`. Adding this to the `printer.cfg` in Mainsail produced no result, as before.

This made me suspect that something was going on other than a faulty SD card or flashing method. So, I tried using a different USB cable in a different port. This time, running `ls -l /dev/serial/by-id/` produced an error message: `ls: cannot access '/dev/serial/by-id/': No such file or directory`

So, we're still not fixed, but at least I was able to produce a different result? What do you make of this?

**Reply by bassamanator** [reply]
Paste your `mcu` section here, surround it with 3 backticks on their own separate lines.

Check `klippy.log` for "Permission denied" error. If that is the case, paste this into terminal: `sudo usermod -a -G tty pi`. [Source](https://www.klipper3d.org/RPi_microcontroller.html#building-the-micro-controller-code). This is likely not the case but worth checking.

**Reply by HemlockMimosa** [reply]
Sure! here is the `mcu` section from the `printer.cfg` in mainsail:

`# NOTE Obtain definition by "ls -l /dev/serial/by-id/"`
'serial: usb-1a86_USB_Serial-if00-port0`
'restart_method: command`

`klippy.log` does not appear to contain any error messages.

**Reply by bassamanator** [reply]
Change your `mcu` section as follows:

```
[mcu]
# NOTE Obtain definition by "ls -l /dev/serial/by-id/"
serial: /dev/serial/by-id/usb-1a86_USB_Serial-if00-port0
restart_method: command
```

**Reply by HemlockMimosa** [reply]
That doesn't appear to have done anything, unfortunately.

**Comment by bassamanator** [comment]
Restart firmware.

**Comment by HemlockMimosa** [comment]
I rebooted klipper, moonraker, and the host in sequence, still not working.

**Comment by HemlockMimosa** [comment]
Does it make a difference how I plug in the RPI? For example, do I plug in
the RPI first, then turn the printer on? Or do I turn the printer on and
then plug in the RPI?

**Comment by bassamanator** [comment]
Paste all logs here in code blocks.

Paste a picture of the error screen.

**Reply by HemlockMimosa** [reply]
Okay! These are huge so I'm going to do them as separate comments. First, the error screen:

![Uploading Screenshot 2023-03-22 215235.png…]()

**Reply by HemlockMimosa** [reply]
that may not have worked, one sec
![Screenshot 2023-03-22 215229](https://user-images.githubusercontent.com/108984157/227080640-44ea7a4c-dd87-4f9f-8b8a-7ecda64c8195.png)

**Reply by HemlockMimosa** [reply]
I'm going to attach the logs as .txt files since they won't fit as code blocks

[klippy server log.txt](https://github.com/bassamanator/Sovol-SV06-firmware/files/11046220/klippy.server.log.txt)
[klippy.log](https://github.com/bassamanator/Sovol-SV06-firmware/files/11046221/klippy.log)
[klippy.log.2023-03-21.txt](https://github.com/bassamanator/Sovol-SV06-firmware/files/11046222/klippy.log.2023-03-21.txt)
[klippy.log.2023-03-22.txt](https://github.com/bassamanator/Sovol-SV06-firmware/files/11046224/klippy.log.2023-03-22.txt)
[mainsail-access.log](https://github.com/bassamanator/Sovol-SV06-firmware/files/11046227/mainsail-access.log)
[mainsail-error.log](https://github.com/bassamanator/Sovol-SV06-firmware/files/11046237/mainsail-error.log)
[moonraker.log](https://github.com/bassamanator/Sovol-SV06-firmware/files/11046230/moonraker.log)
[moonraker.log.2023-03-21.txt](https://github.com/bassamanator/Sovol-SV06-firmware/files/11046231/moonraker.log.2023-03-21.txt)
[moonraker.log.2023-03-22.txt](https://github.com/bassamanator/Sovol-SV06-firmware/files/11046232/moonraker.log.2023-03-22.txt)

**Reply by bassamanator** [reply]
The error is in moonraker.conf...

**Reply by HemlockMimosa** [reply]
ahhh thank you for pointing that out! Do you know what I need to do in order to fix it?

**Reply by bassamanator** [reply]
Paste moonraker.conf.

**Reply by HemlockMimosa** [reply]
[moonraker conf.txt](https://github.com/bassamanator/Sovol-SV06-firmware/files/11051240/moonraker.conf.txt)

Here you go! I'm sorry this has been such a pain in the ass.

**Reply by bassamanator** [reply]
No worries. In moonraker.conf, change the following line:
```
klippy_uds_address: /home/pi/printer_data/comms/klippy.sock
```
to
```
klippy_uds_address: /home/andre/printer_data/comms/klippy.sock
```

Also edit `cfgs/misc-macros.cfg` and change 
```
[virtual_sdcard]
path: /home/pi/printer_data/gcodes
```
to
```
[virtual_sdcard]
path: /home/andre/printer_data/gcodes
```

**Reply by HemlockMimosa** [reply]
It feels like we're really close now, because I got to a new error message.

`mcu 'mcu': Unable to connect`
`Once the underlying issue is corrected, use the
"FIRMWARE_RESTART" command to reset the firmware, reload the
config, and restart the host software.`
`Error configuring printer`

I'm attaching the latest logs:

[moonraker (3).log](https://github.com/bassamanator/Sovol-SV06-firmware/files/11053307/moonraker.3.log)
[klippy (6).log](https://github.com/bassamanator/Sovol-SV06-firmware/files/11053309/klippy.6.log)

**Reply by bassamanator** [reply]
`Firmware_restart`?

**Reply by Trilis29** [reply]
Hi, I think I have same proble. Everything configured as on tutorial still can't connect.

**Comment by SporePrint** [comment]
I know this is an old post, but I'm running into some issues similar to comment #6; 
I've printed loads with my SV06 using bassamanator's configs as a starting point(you're an mvp homie, thank you for your work); then my printer sat for about a year. I reinstalled linux on my host due to auth issues, and am back to the point of calibrating/fine-tuning my printer.

For some reason, I'm having a _lot_ of trouble getting my host to connect to my printer upon initial boot up. I have to sit and fiddle, going back and forth between my printer and my remote klipper interface. I "restart" klipper via the interface, and have to toggle the printer on and off a few times before klipper errors out in order to connect, sometimes needing to restart klipper multiple times. If I power up my printer and then restart klipper, it seems that my printer times out/freezes before klipper attempts to connect. Once it connects, I'm good to go for the duration of both machines staying online, but that initial connection seems so finnicky for some reason, and I don't remember it to be so.

I'm posting here mostly because I thought it may be relevant to anyone experiencing issues similar to above - just because your setup doesn't smoothly boot up first try doesn't necessarily mean you didn't flash properly or have your config setup properly, it really seems to just be finnicky to boot up!

If you have suggestions on why this may be or how to resolve it, I'd love to hear them!

**Reply by bassamanator** [reply]
What device is hosting klipper (rpi, btt pad, etc.)?

**Reply by SporePrint** [reply]
I'm currently running on an older laptop, I thought I had included that in my comment! I was running on a rpi3 but couldn't get past the power supply issues so moved on to a retired laptop running Mint. Once I get through to connection song and dance I'm set until I power off my printer; it's robust to `restarts` within the interface. This leads me to rule out my physical connections. It almost seems as though klipper can't send a `firmware_restart` command until after the first connection is successful, now that I think a little more about it.

I use the Mainsail interface on a separate windows laptop, though I feel that's irrelevant as I don't run into many issues between host and interface other than sleeping tabs.

**Reply by bassamanator** [reply]
So you're having connection issues only when you A) boot the host, or when you B)power on the printer?

In the case of A), the klipper service (which can be checked via `systemctl status klipper.service` in the host's terminal) does take a bit to start up. There are other services as well, `moonraker.service` for example, that take some time as well. To see if there are *actual* connection issues, keep the printer on, then:
1. restart the host device.
2. wait 5 minutes, or rather make it 10 minutes because the laptop might be especially slow.
3. goto the klipper dashboard, and hit firmware restart *once*.

If without further intervention you're able to get to an error free klipper dashboard, you in fact have no connection issues.

**Reply by SporePrint** [reply]
My host stays on indefinitely due to multi use.
 
For soe clarity, let me explain in detail. This is mostly an issue when starting my day, I walk in with my coffee, flick on the printer, then cross the room to my workstation where I wake my pc and load Mainsail in the browser where it initializes that remote connection, then nothing happens. Klipper will eventually error out with an mcu connection error(can't remember which one at the moment but it's the same one it gives if the printer isn't toggled on). `restart` and `firmware_restart` have no affect on anything. If I leave it be, it times out. Restarting any of the services from the `power` menu in mainsail has no affect on anything, everything will boot up as normal, I get the blue "attempting connection" screen, and then it times out. 

The only way to boot up is to turn the printer on, restart klipper, turn the printer off(seemingly after a certain amount of time), then back on immediately, and restart klipper again. This reliably connects everything, except sometimes I get (I think) the mcu clock error and need to start over. I don't remember that to be the case, I remember being able to flick the power switch on my printer and then go to my workstation to start my warmup routines. 

Is it possible that in my reinstallation of everything I've missed a tiny bit of config somewhere that keeps klipper searching longer, or something along those lines? I do not have nearly enough software engineering experience to dig through and know what everything does lol

**Reply by bassamanator** [reply]
> The only way to boot up is to turn the printer on, restart klipper, turn the printer off(seemingly after a certain amount of time), then back on immediately, and restart klipper again.

How are you restarting Klipper? The Klipper host is always up you say, are you able to SSH into it in the morning?

**Reply by SporePrint** [reply]
I am able to ssh into the host in the morning, I always restart klipper from the Mainsail interface. The host never gets touched, it's screen is disabled and all power saving disabled, tucked in a shelf under the printer. When I wake my remote PC, I navigate to the still-open Mainsail browser tab and get a 2-5 second "Initializing" screen before the mainsail interface with blue "attempting connection" appears. Nothing gets shut down in any way other than the printer with the physical switch, as I remember the SV06 stock board doesn't allow for remote shutdown.

**Reply by bassamanator** [reply]
Next morning:
1. turn on the printer
2. open a new browser and go to mainsail
3. give it a minute
4. if 2 still doesn't open up mainsail error free, try 2 but on a smartphone
5. if you're still unable to connect, ssh into the host and run `systemctl status moonraker.service klipper.service` (`q` exits the output you will see). Paste output in a codeblock here.

And to answer a previous question, no, there's nothing you could have missed in the initial setup to cause this.

**Reply by SporePrint** [reply]
I followed your directions; 
- After no change for 5 minutes I did a klipper `firmware_restart` from the error message on the Mainsail dashboard. 
- No change again for 5 minutes so another `firmware_restart` and got up to toggle printer's power. I think I was too slow getting to the printer, because it attempted to connect but glitched - my SV06 screen was full of random pixels(I'm sure it wasn't random, but I couldn't interpret what was going on). 
- Toggled printer power again and nothing.
- Another `firmware_restart` and rushing over to my printer to toggle the power has me connected and running as normal again, homing my printer and heating my chamber as I type this. 

My prints turn out great(calibration needs a little tuning still), and I have no "bottlenecks" even when cranking the print speeds up, the ONLY issue is that initial connection. To reiterate, once that connection is made I can edit configs and `save and restart`, I can walk away for a few hours leaving everything running but idle, I can run multiple prints in a row. I just cannot reliably connect easily when powering on the printer.

As I'd mentioned this is more of an annoyance than anything, once I'm connected there are almost no hurdles. This is what lead me to post about this "issue", so that others using the mediocre SV06 stock mainboard would know to fiddle a little more before assuming they've failed a setup step. If there's a solution, I'm definitely game to figure it out - anything that makes a workflow smoother and more comfortable is a win in my opinion - but I'm starting to feel like this really is just a side-effect of whatever shortcuts they took making the mainboard as cheap as possible(for example not being able to shut down the printer via a command). I don't remember all of the advantages of upgrading to an aftermarket mainboard, but I'm thinking these issues would probably be resolved if I did.

I decided to run the `systemctl status moonraker.service klipper.service` both before doing anything and after successfully connecting, see the output below;
``` 
● moonraker.service - API Server for Klipper SV1
     Loaded: loaded (/etc/systemd/system/moonraker.service; enabled; vendor pre>
     Active: active (running) since Wed 2024-12-11 16:04:51 AST; 6 days ago
       Docs: https://moonraker.readthedocs.io/
   Main PID: 950 (python)
      Tasks: 8 (limit: 6443)
     Memory: 100.8M
        CPU: 4h 6min 47.553s
     CGroup: /system.slice/moonraker.service
             └─950 /home/fungi/moonraker-env/bin/python /home/fungi/moonraker/m>

Dec 18 00:50:03 klipper-control python[950]: System Time: 1734497395.524622, Us>
Dec 18 00:50:03 klipper-control python[950]: System Time: 1734497396.525494, Us>
Dec 18 00:50:03 klipper-control python[950]: System Time: 1734497397.526794, Us>
Dec 18 00:50:03 klipper-control python[950]: System Time: 1734497398.529386, Us>
Dec 18 00:50:03 klipper-control python[950]: System Time: 1734497399.534216, Us>
Dec 18 00:50:03 klipper-control python[950]: System Time: 1734497400.535590, Us>
Dec 18 00:50:03 klipper-control python[950]: System Time: 1734497401.536667, Us>
Dec 18 00:50:03 klipper-control python[950]: System Time: 1734497402.537659, Us>
Dec 18 00:50:03 klipper-control python[950]: System Time: 1734497403.538595, Us>
Dec 18 00:50:03 klipper-control python[950]: CPU Temperature: None

● klipper.service - Klipper 3D Printer Firmware SV1
     Loaded: loaded (/etc/systemd/system/klipper.service; enabled; vendor prese>
     Active: active (running) since Mon 2024-12-16 15:35:02 AST; 1 day 18h ago
       Docs: https://www.klipper3d.org/
   Main PID: 123389 (python)
      Tasks: 2 (limit: 6443)
     Memory: 47.6M
        CPU: 1h 32min 41.093s
     CGroup: /system.slice/klipper.service
             └─123389 /home/fungi/klippy-env/bin/python /home/fungi/klipper/kli>

Dec 16 15:35:02 klipper-control systemd[1]: Started Klipper 3D Printer Firmware>
lines 12-34/34 (END)

=================================================================================

● moonraker.service - API Server for Klipper SV1
     Loaded: loaded (/etc/systemd/system/moonraker.service; enabled; vendor pre>
     Active: active (running) since Wed 2024-12-11 16:04:51 AST; 6 days ago
       Docs: https://moonraker.readthedocs.io/
   Main PID: 950 (python)
      Tasks: 8 (limit: 6443)
     Memory: 100.9M
        CPU: 4h 7min 29.702s
     CGroup: /system.slice/moonraker.service
             └─950 /home/fungi/moonraker-env/bin/python /home/fungi/moonraker/m>

Dec 18 10:00:43 klipper-control python[950]: Error configuring printer
Dec 18 10:03:05 klipper-control python[950]: [klippy_connection.py:_on_connecti>
Dec 18 10:03:06 klipper-control python[950]: [klippy_connection.py:_do_connect(>
Dec 18 10:03:44 klipper-control python[950]: [klippy_connection.py:_on_connecti>
Dec 18 10:03:45 klipper-control python[950]: [klippy_connection.py:_do_connect(>
Dec 18 10:03:59 klipper-control python[950]: [klippy_connection.py:_request_ini>
Dec 18 10:03:59 klipper-control python[950]: [klippy_connection.py:_request_ini>
Dec 18 10:03:59 klipper-control python[950]: [job_state.py:_handle_started()] ->
Dec 18 10:03:59 klipper-control python[950]: [klippy_connection.py:_check_ready>
Dec 18 10:04:00 klipper-control python[950]: [data_store.py:_init_sensors()] - >

● klipper.service - Klipper 3D Printer Firmware SV1
     Loaded: loaded (/etc/systemd/system/klipper.service; enabled; vendor prese>
     Active: active (running) since Mon 2024-12-16 15:35:02 AST; 1 day 18h ago
       Docs: https://www.klipper3d.org/
   Main PID: 123389 (python)
      Tasks: 4 (limit: 6443)
     Memory: 39.5M
        CPU: 1h 33min 39.981s
     CGroup: /system.slice/klipper.service
             └─123389 /home/fungi/klippy-env/bin/python /home/fungi/klipper/kli>

Dec 16 15:35:02 klipper-control systemd[1]: Started Klipper 3D Printer Firmware>
lines 1-12/12 (END)

**Reply by bassamanator** [reply]
> mediocre SV06 stock mainboard

> I'm starting to feel like this really is just a side-effect of whatever shortcuts they took making the mainboard as cheap as possible(for example not being able to shut down the printer via a command)

The board is perfectly adequate for the job at hand. It is a **rarity** to have the built-in functionality of being able to power off the printer via a command. You're frustrated, and are overstating the issue here, which is understandable. And this is a very annoying issue, because you're having to fiddle with the `powering on` process, which should be the easiest thing of all. Moreover, I would say that the true shortcoming of this board is the fact that the mcu fan and the part cooling fan are linked together, but let's not get distracted.

Next 'wake up' cycle when the mainsail dashboard shows an error state, I want you to see if you can detect the mcu via the `ls` command [you used](https://github.com/bassamanator/Sovol-SV06-firmware?tab=readme-ov-file#adjust-configuration-with-mcu-path) to get the path initially.

Lastly, did you try to get to the dashboard via a different device? Are you using the browser on the Klipper host?

EDIT: The services are running just fine.

**Reply by SporePrint** [reply]
> The board is perfectly adequate for the job at hand.

It absolutely is, I suppose I was being a little harsh on it. As mentioned I've been out of the loop from the 3D printing community for a while, and I don't remember everything I'd learned on my previous journey through setting up this printer and learning the ins and outs of 3D printing in general, but especially this printer and klipper. I do remember than many people mentioned wanting to upgrade the mainboard for several reasons, and the ability to remotely power on/off the printer was the one that stuck in my head. I don't even remember what the options for upgrades were, `Btt Skr mini e3 v3` is familiar after a quick google search. Perhaps my desire to upgrade was based on issues I was running into prior, when trying to use the rpi3 as a host. I know I was, and still am, annoyed that flashing the firmware can _only_ be done via microSD; I'm very jealous of those who can do that via usb aas it can be a real pain with this printer(took me an entire week in my first foray into klipper)!

I will see about finding the mcu via terminal tomorrow, I suspect I won't be able to but we shall see. Thanks for the reminder, I disregarded this as a troubleshooting step!

I did try the dashboard in both a new browser on my pc, and on my mobile. I'm not using anything on the host, the browser has never even been opened on that laptop since reinstalling Mint. I simply installed Mint, then used the terminal to update > upgrade > install git > openssh then it was closed up and packed away as a headless unit. I did the `KIAUH` install remotely and everything since. 

If it helps to clarify a little, the host is permanently "awake" and I can ssh in at any time to check services, reboot, etc. I'm also able to use the Mainsail interface for anyhting other than connecting to the mcu; I can upload gcode, edit configs, and whatever else. It is in no way a "dead instance", and in my little connection "song and dance" I never need to refresh the browser. It just seems like for some reason klipper doesnt know where to send a `firmware_restart` command until after the printer has been cycled several times, potentially "waking" the mcu connection(?). Typing this out made me wonder if maybe there's somewhere(other than printer.cfg) that I need to include my mcu path?

**Reply by bassamanator** [reply]
> Typing this out made me wonder if maybe there's somewhere(other than printer.cfg) that I need to include my mcu path?

Nope, just in `printer.cfg`.

Let's see what happens tomorrow. Keep me posted.

**Reply by SporePrint** [reply]
Is it possible my issues are caused by;
- Before connection song and dance, while printer powered on,
`fungi@klipper-control:~$ ls /dev/seial/by-id
ls: cannot access '/dev/seial/by-id': No such file or directory`
- After connecting successfully,
`fungi@klipper-control:~$ ls /dev/serial/by-id/
usb-1a86_USB_Serial-if00-port0`

**Reply by bassamanator** [reply]
Progress! 

The printer was powered on, I'm sure. Let me think on this.

**Reply by bassamanator** [reply]
Out of curiosity, how long is the USB cable that connects the printer to the host?

**Reply by SporePrint** [reply]
I believe it's 180-300cm; it's a cord from an old xbox controller. When I was using the rpi it was the only one in my house that gave a reliable connection(neither side loose in the connector, power+data, robust/low-gauge), so I've trusted it for a long while.

**Reply by bassamanator** [reply]
Considering that the `mcu` was not found via `ls` upon 'waking', and that everything else is looking as it should, I believe the *length* of the cable is a factor.

**Reply by SporePrint** [reply]
So, are you thinking it's searching for a fingerprint of a sleeping serial connection, and until that mcu is awake the "search signal" is too weak to detect it because of voltage drop across the cable, but once the mcu cycles it can see it because a signal is coming from both ends? Interesting, though this is the same cable I used last year with what I remember to be less connection drama. 

I'll have to wait till after the holidays to try new cables, as I mentioned any others in the house are insufficient and I don't have the means to purchase new at the moment

**Reply by bassamanator** [reply]
Yes, something along those lines.

I too would hate to buy yet another USB cable. Perhaps see if a friend/neighbour/colleague is able to let you borrow a more standard length cable for ~24 hours. This way you can cross this potential issue off the `to check` list without waste.
